src/shared_modules/content_manager/src/components/CtiDownloader.hpp

```diff
@@ -85,22 +85,56 @@ class CtiDownloader : public AbstractHandler<std::shared_ptr<UpdaterContext>>
      */
     CtiBaseParameters getCtiBaseParameters(const std::string& ctiURL)
     {
-        CtiBaseParameters parameters;
+        nlohmann::json rawMetadata;
 
         // Routine that stores the necessary parameters.
-        const auto onSuccess {[&parameters](const std::string& response)
+        const auto onSuccess {[&rawMetadata](const std::string& response)
                               {
-                                  const auto responseData = nlohmann::json::parse(response).at("data");
+                                  logDebug2(WM_CONTENTUPDATER, "CTI raw metadata: '%s'", response.c_str());
+
+                                  if (!nlohmann::json::accept(response))
+                                  {
+                                      throw std::runtime_error {"Invalid CTI metadata format"};
+                                  }
 
-                                  parameters.lastOffset = responseData.at("last_offset").get<int>();
-                                  parameters.lastSnapshotLink =
-                                      responseData.at("last_snapshot_link").get<std::string>();
-                                  parameters.lastSnapshotOffset = responseData.at("last_snapshot_offset").get<int>();
+                                  auto responseJSON = nlohmann::json::parse(response);
+                                  if (!responseJSON.contains("data"))
+                                  {
+                                      throw std::runtime_error {"No 'data' field in CTI metadata"};
+                                  }
+
+                                  rawMetadata = std::move(responseJSON.at("data"));
                               }};
 
         // Make a get request to the API to get the consumer offset.
         performQueryWithRetry(ctiURL, onSuccess);
 
+        // Return if interrupted.
+        if (m_spUpdaterContext->spUpdaterBaseContext->spStopCondition->check())
+        {
+            return CtiBaseParameters();
+        }
+
+        // Validate metadata.
+        for (const std::string& key : {"last_offset", "last_snapshot_link", "last_snapshot_offset"})
+        {
+            if (!rawMetadata.contains(key))
+            {
+                throw std::runtime_error {"Missing CTI metadata key: " + key};
+            }
+
+            const auto& data {rawMetadata.at(key)};
+            if (data.is_null() || (data.is_string() && data.get_ref<const std::string&>().empty()))
+            {
+                throw std::runtime_error {"Null or empty CTI metadata value for key: " + key};
```

Comment:
This exception stops the execution of the orchestration and does not allow the offsets download stage to be executed. It may happen that a consumer only publishes offsets and not snapshot, so we must be able to process both contents separately.

Please make the necessary changes to allow offsets processing even if there are no snapshots created.

Reply:
Good. I'll need one more day for implementing this.

Reply:
This is done!